### PR TITLE
Added FE of Meme merge

### PR DESCRIPTION
**Pull request by Rohan749**
## Release Notes

- **New Features**
	- Introduced several new components for managing token merging and claiming processes, including `ClaimCompleteModal`, `ClaimProcessingModal`, `MemeClaimForm`, `MemeMergeForm`, `MergeCompleteModal`, `MergeProcessingModal`, `MergePriceChart`, and `ChaosRewardCard`.
	- Added functionality for users to view transaction details and manage claims through modals.
	- Implemented a new meme merging page with server-side rendering for dynamic configuration retrieval.
	- Added new tokens (`goodle` and `fomo`) to the merge options, expanding user choices.

- **Improvements**
	- Enhanced input handling in forms with new properties for better user experience.
	- Updated conditional rendering logic for improved layout handling based on URL paths.

- **Bug Fixes**
	- Adjusted styling in merge form components to ensure consistent UI appearance.

### CODE REVIEW

**Comment by xBi0** [comment]
ready for testing and "integration" 

waits ratio and mechanics confirmation to go live

File: src/components/ui/Merge/Goodle/MergeCompleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    </Typography>
+                    <Typography size="md" className="font-bold">
+                      0.0008
+                    </Typography>
+                  </>    
+              }
+            </div>
+          </div>
+          <Image
+            src="/videos/bro-merge-animation.gif"
```
Comment (mmrz) [inline src/components/ui/Merge/Goodle/MergeCompleteModal.tsx:76]: I suppose we need to use a different animation here.

File: src/components/ui/Merge/Goodle/MergeCompleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            <Image
+              src="/img/tokens/goodle.svg"
+              width={17}
+              height={17}
+              alt="GOODLE logo"
+            />
+
+            <Typography size="md" fw={600}>
+              {/* ≈ {dn.format([props.ratio, 2], { locale: "en" })} */}
+                5,000
```
Comment (mmrz) [inline src/components/ui/Merge/Goodle/MergeCompleteModal.tsx:47]: Static data. It should come from SC.

File: src/components/ui/Merge/Goodle/MergeCompleteModal.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+              alt="WEWE logo"
+            />
+            <div className="flex flex-col">
+              {
+                   <>
+                    <Typography size="sm" className="text_light_gray">
+                      CLAIMED
+                    </Typography>
+                    <Typography size="md" className="font-bold">
+                      0.0008
```
Comment (mmrz) [inline src/components/ui/Merge/Goodle/MergeCompleteModal.tsx:69]: Static data. It should come from SC events.

File: src/constants/addresses.ts [hunk 4]
```diff
@@ -24,4 +24,5 @@ export const CONTRACT_ADDRESSES = {
     "0x6f71796114b9cdaef29a801bc5cdbcb561990eeb" as `0x${string}`,
   nonFungiblePositionManagerAddress:
     "0x03a520b32C04BF3bEEf7BEb72E919cf822Ed34f1" as `0x${string}`,
+  goodle: "0x40312F66C4362FA165607E28D3f812b463761803" as `0x${string}`
```
Comment (mmrz) [inline src/constants/addresses.ts:27]: The name goodle is misleading. Can be renamed to something like: `goodleEater`

File: src/lib/abis/GoodleABI.ts [hunk 5]
```diff
@@ -0,0 +1,393 @@
+const GoodleAbi = [
```
Comment (mmrz) [inline src/lib/abis/GoodleABI.ts:1]: nit: GoodleAbi is misleading. Can be renamed to: GoodleEaterAbi

File: src/pages/merge/goodle/index.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import { CONTRACT_ADDRESSES } from '~/constants'
+import { useEaterRate } from '~/hooks/useEater'
+import { useTokenBalance } from '~/hooks/useTokenBalance'
+import * as dn from "dnum";
+
+const GoodleMergePage = () => {
+
+  const { rate } = useEaterRate(CONTRACT_ADDRESSES.goodle);
+
+  console.log("RATE:", dn.format([rate, 2], { locale: "en" }))
```
Comment (mmrz) [inline src/pages/merge/goodle/index.tsx:15]: extra console.log

File: src/pages/merge/goodle/index.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+              <li>Fixed Rate of <strong>1 $GOODLE to 36.45 $WEWE</strong></li>
+            </ul>
+          </div>
+        </Card>
+        <Card>
+        <div id="dexscreener-embed" className='h-[25rem] lg:h-[40rem]'>
+          <iframe 
+            className='w-full h-full' 
+            src="https://dexscreener.com/base/0x5E9BB3d7682A9537DB831060176C4247aB80D1Ec?embed=1&theme=dark"> 
+          </iframe>
```
Comment (mmrz) [inline src/pages/merge/goodle/index.tsx:53]: If it's not working, replace it with a text or blank div for now.

**Pull request merged**